### PR TITLE
New genome-env option

### PR DESCRIPTION
**Pull request by apregier**
I'm not sure if this is a good idea or not, but it seemed useful to us, for example for debugging model tests

### CODE REVIEW

**Comment by acoffman** [comment]
I may not be the best person to review the shell script itself, but this definitely does seem like a useful feature, and it looks to be implemented correctly!

+1

File: bin/genome-env [hunk 1]
```diff
@@ -89,7 +102,14 @@ function main {
...
-        log.info "Test DB Snapshot: $DB_SNAPSHOT_NAME"
+        if test -n "$DB_ID"
+        then
+            log.info "Using existing test DB"
+            log.info "Test DB Host: ${TESTDBSERVER_DB_HOST}"
+            log.info "Test DB Port: ${TESTDBSERVER_DB_PORT}"
+        else
+            log.info "Test DB Snapshot: $DB_SNAPSHOT_NAME"
+        fi
         log.info "Test DB Name: $TESTDBSERVER_DB_NAME"
```
Comment (ebelter) [inline bin/genome-env]: Why not merge this if/else block with the one above?

Comment (apregier) [inline bin/genome-env]: I'm not sure I know what you mean, but if you mean the block starting at line 80, I guess the reason is to maintain the order of the log messages.

Comment (ebelter) [inline bin/genome-env]: That is the block I was siting.

**Comment by nnutter** [comment]
This either may not be needed or will be insufficient as is.  How do you intend on debugging model tests?  If you mean we get a failure and you want to be able to enter that ephemeral database with this then you'll also need an option to disable removing the ephemeral database when the `genome-env` process ends.  If you just want to be able to manually run a model tests and debug then this isn't necessary as you can just drop into a shell with `genome-env`, run the model test, and then debug as needed.

**Comment by apregier** [comment]
Yes, I was thinking of looking at failed model tests from Jenkins, or even just checking on ones that are currently running.  I did include a check before deleting the db to see if it was loaded with the new option (line 133).

**Comment by nnutter** [comment]
I meant that the model tests are going to delete their ephemeral database when they complete.  (Though I imagine there's probably going to be some resulting inconsistency if someone's connected to the database when they try to do that.)

**Comment by apregier** [comment]
Oh ok - yes, I was thinking we need that option too!  Is it true that the dbs expire after a month anyway?  It seems like it would be useful to have them for at least a couple days in the case of failed builds, and in the case of succeeded builds we could explicitly delete them.

**Comment by nnutter** [comment]
The current default is 7 days.  If that needs to be changed or, more likely, configurable we'll need to add that to `test-db`.

**Comment by apregier** [comment]
7 days seems reasonable to me - or do you think it will be too much of a resource hog to keep the dbs for model tests around that long?  I pushed a commit that adds the option of persisting the db beyond the life of the shell.

File: bin/genome-env [hunk 2]
```diff
@@ -541,6 +585,8 @@ default_value TESTDBSERVER_DB_USER "genome"
 default_value TESTDBSERVER_DB_PASS "mypassword"
 default_value TESTDBSERVER_URL "https://apipe-test-db.gsc.wustl.edu"
 default_value GE_ENV_NAME "genome-env"
+default_value TESTDBSERVER_DB_HOST "apipe-test-db.gsc.wustl.edu"
+default_value TESTDBSERVER_DB_PORT "5432"
```
Comment (nnutter) [inline bin/genome-env]: `TESTDBSERVER_DB_HOST` and `TESTDBSERVER_DB_PORT` defaults shouldn't be in here I don't think.  They get set by `$TESTDB database create --bash` and is information that should either come from the developer, for `-i`, or from `test-db`.

File: bin/genome-env [hunk 3]
```diff
@@ -89,7 +104,14 @@ function main {
 
     if test -z "$DISABLE_DB"
     then
-        log.info "Test DB Snapshot: $DB_SNAPSHOT_NAME"
+        if test -n "$DB_ID"
+        then
+            log.info "Using existing test DB"
+            log.info "Test DB Host: ${TESTDBSERVER_DB_HOST}"
+            log.info "Test DB Port: ${TESTDBSERVER_DB_PORT}"
```
Comment (nnutter) [inline bin/genome-env]: I think it would be fine to always show the host and port, i.e. move them down with "Test DB Name".

**Comment by nnutter** [comment]
> 7 days seems reasonable to me - or do you think it will be too much of a resource hog to keep the dbs for model tests around that long?

We'll just have to monitor disk usage on `apipe-test-db` to see.

File: bin/genome-env (hunk 2 above)
Comment (apregier) [inline bin/genome-env]: I agree - those were from before when we were trying to just pass in the db name with -i; I should have deleted them!

File: bin/genome-env (hunk 3 above)
Comment (apregier) [inline bin/genome-env]: Ok, done!

**Comment by nnutter** [comment]
:+1:

**Pull request merged**